metals/src/main/scala/scala/meta/internal/metals/ScalafixProvider.scala

```diff
@@ -35,52 +39,76 @@ case class ScalafixProvider(
-      } yield {
-        val scalacOption =
-          if (scalaBinaryVersion == "2.13") "-Wunused:imports"
-          else "-Ywarn-unused-import"
-
-        api
-          .newArguments()
-          .withScalaVersion(scalaVersion)
-          .withClasspath(classpath)
-          .withToolClasspath(urlClassLoaderWithExternalRule)
-          .withConfig(scalafixConf.asJava)
-          .withRules(List(organizeImportRuleName).asJava)
-          .withPaths(List(file.toNIO).asJava)
-          .withSourceroot(workspace.toNIO)
-          .withScalacOptions(Collections.singletonList(scalacOption))
-          .evaluate()
-      }
+      compilations.compilationFinished(file).map { _ =>
+        val scalafixConfPath = workspace.resolve(scalafixConfigPath)
+        val scalafixConf =
+          if (scalafixConfPath.isFile) Some(scalafixConfPath.toNIO)
+          else None
+        val scalafixEvaluation = for {
+          api <- getScalafix(scalaBinaryVersion)
+          urlClassLoaderWithExternalRule <- getRuleClassLoader(
+            scalaBinaryVersion,
+            api.getClass.getClassLoader
+          )
+        } yield {
+          val scalacOption =
+            if (scalaBinaryVersion == "2.13") "-Wunused:imports"
+            else "-Ywarn-unused-import"
+
+          api
+            .newArguments()
+            .withScalaVersion(scalaVersion)
+            .withClasspath(classpath)
+            .withToolClasspath(urlClassLoaderWithExternalRule)
+            .withConfig(scalafixConf.asJava)
+            .withRules(List(organizeImportRuleName).asJava)
+            .withPaths(List(file.toNIO).asJava)
+            .withSourceroot(workspace.toNIO)
+            .withScalacOptions(Collections.singletonList(scalacOption))
+            .evaluate()
+        }
+        val edits = for {
+          results <- scalafixEvaluation
+          if results.isSuccessful()
+          fileEvaluation <- results.getFileEvaluations().headOption
+          patches <- fileEvaluation.previewPatches().asScala
+        } yield textEditsFrom(patches, fileInput)
+
+        if (!scalafixEvaluation.exists(_.isSuccessful())) {
+          val scalafixError =
+            scalafixEvaluation.flatMap(_.getMessageError().asScala)
+
+          val message = scalafixError.getOrElse {
+            """|Unexpected error while running scalafix.
+               |Semanticdb might have been stale, which 
+               |would require successful compilation to be created""".stripMargin
+          }
+          languageClient.showMessage(warningMessage(message))
 
-      val newFileContentOpt = scalafixEvaluation
-        .flatMap(result => result.getFileEvaluations.headOption)
-        .flatMap(_.previewPatches().asScala)
-      newFileContentOpt.map(textEditsFrom(_, fileInput)).getOrElse(Nil)
+        }
+
+        edits.getOrElse(Nil)
+      }
     }
   }
 
+  private def warningMessage(
+      message: String
+  ): MessageParams = {
+    new MessageParams(MessageType.Error, message)
+  }
```

Comment:
Sort of a nit,but we call this method `warningMessage`, but we give the message an `Error` type. Should we align those?

Reply:
changed the name and it now reflects the fact it's an error.